zulipterminal/ui_tools/views.py

```diff
@@ -779,6 +779,35 @@ def __init__(self, controller: Any, widgets: List[Any],
         self.log = urwid.SimpleFocusListWalker(widgets)
         super().__init__(self.log)
 
+    @staticmethod
+    def table_widths(contents: PopUpViewTableContent, popup_title_len: int,
+                     dividechars: int=2) -> Tuple[int, List[int]]:
+        """
+        Returns a tuple that contains the required width for the popup and a
+        list that has column widths.
+        """
+        # Add 4 (for 2 Unicode characters on either side) to the popup title
+        # length to make sure that the title gets displayed even when the
+        # content or the category is shorter than the title length (+4 Unicode
+        # characters).
+        title_width = popup_title_len + 4
+
+        category_width = 0
+        text_width = 0
+        strip_widths = []
+        for category, content in contents:
+            category_width = max(category_width, len(category))
+            for row in content:
+                if isinstance(row, str):
+                    text_width = max(text_width, len(row))
+                if isinstance(row, tuple):
+                    strip_widths.append(list(map(len, row)))
+        column_widths = [max(width) for width in zip(*strip_widths)]
+
+        popup_width = max(title_width, category_width, text_width,
+                          sum(column_widths) + dividechars)
```

Comment:
This is actually different behavior - arguably better and more general, but again not just a refactor!
(some of that comes from the other extra feature, ie. the `text_width`, of course)